### PR TITLE
Enhance minikube getting started page

### PR DESCRIPTION
**Pull request by sdake**
Several errors with the minikube page under more intense review. These
errors were fixed.  Several warnings were added specifically about not
specifying enough memory.

Fixes: 4923

### CODE REVIEW

**Comment by sebgoa** [comment]
There has to be a set of values that can be used to lower this memory requirement in a development setup like minikube.

I was pointed to:

https://github.com/weaveworks/flagger/blob/master/test/e2e-istio-values.yaml

**Comment by sdake** [comment]
@sebgoa our minimal istio deployment is specified by https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values-istio-minimal.yaml although we don't distribute a pre-rendered yaml of this configuration.  As a project, we struggle with operators who deploy the `istio-demo-auth.yaml` we do pre-render.  We make it too easy to deploy this particular manifest, and this manifest performs very poorly in a production environment.

Cheers
-steve

File: content/docs/setup/kubernetes/platform-setup/minikube/index.md [hunk 1]
```diff
@@ -1,46 +1,87 @@
...
-   [Minikube](https://kubernetes.io/docs/setup/minikube/), version **1.0.0 or
+   [minikube](https://kubernetes.io/docs/setup/minikube/), version **1.1.1 or
    later**, and a
-   [Minikube Hypervisor Driver](https://kubernetes.io/docs/tasks/tools/install-minikube/#install-a-hypervisor).
+   [minikube hypervisor driver](https://kubernetes.io/docs/tasks/tools/install-minikube/#install-a-hypervisor).
 
     {{< tip >}}
-    Set your Minikube Hypervisor driver.  For example if you installed the KVM hypervisor, set the vm-driver
-    within the Minikube configuration using the following command:
+    Set your minikube Hypervisor driver.  For example if you installed the KVM hypervisor, set the vm-driver
```
Comment (ericvn) [inline content/docs/setup/kubernetes/platform-setup/minikube/index.md]: lower case Hypervisor here as well?

Comment (sdake) [reply content/docs/setup/kubernetes/platform-setup/minikube/index.md]: thanks done.

File: content/docs/setup/kubernetes/platform-setup/minikube/index.md [hunk 2]
```diff
@@ -52,14 +53,14 @@ Administrative privileges are required to run minikube.
...
+    {{< text bash >}}
     $ minikube ssh
     $ top
     GiB Mem : 12.4/15.7
-    {{< /text_hack >}}
+    {{< /text >}}
 
-    This shows 12.4GiB used of 15.7 GiB in the virtual machine.  This data was generated with
-    VMWare Fusion hypervisor on a Macbook Pro 13" running Istio 1.2 with bookinfo installed.
+    This shows 12.4GiB used of 15.7 GiB RAMin the virtual machine.  This data was generated with
```
Comment (ericvn) [inline content/docs/setup/kubernetes/platform-setup/minikube/index.md]: Spelling will probably get this, RAMin => RAM in.

Comment (sdake) [reply content/docs/setup/kubernetes/platform-setup/minikube/index.md]: thanks @ericvn .

**Comment by sebgoa** [comment]
fwiw, I patched the knative lean installation to set the pilot memory request to 128Mi:

https://github.com/knative/serving/blob/master/third_party/istio-1.1.7/download-istio.sh#L39

This brings it down from the default 2Gi.

for the record I still believe that asking people to run minikube with 16 Gi RAM to test is Istio is way too high and unpractical. You would be better off removing any minikube instructions.

**Comment by sdake** [comment]
> fwiw, I patched the knative lean installation to set the pilot memory request to 128Mi:
> 
> https://github.com/knative/serving/blob/master/third_party/istio-1.1.7/download-istio.sh#L39
> 
> This brings it down from the default 2Gi.
> 
> for the record I still believe that asking people to run minikube with 16 Gi RAM to test is Istio is way too high and unpractical. You would be better off removing any minikube instructions.

Such a change makes sense for a lean demo install.  This documentation was tested with this set of memory requests (100mb): https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values-istio-demo-common.yaml#L27

**Pull request merged**